src/net/tls.cc

```diff
@@ -710,6 +718,11 @@ void tls::credentials_builder::set_priority_string(const sstring& prio) {
 
 void tls::credentials_builder::set_session_resume_mode(session_resume_mode m) {
     _session_resume_mode = m;
+    if (m != session_resume_mode::NONE) {
+        gnutls_datum key;
+        gnutls_session_ticket_key_generate(&key);
```

Comment:
might want to check the return value of `gnutls_session_ticket_key_generate()`? as, in theory, it could fail by returning a non-zero value. see https://man7.org/linux/man-pages/man3/gnutls_session_ticket_key_generate.3.html and https://github.com/gnutls/gnutls/blob/94e01a704f307a83ccb7799d1c2626f9635f2b5c/lib/ext/session_ticket.c#L516 .

Reply:
True. Can OOM. Added check to both calls.